flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionTest.java

```diff
@@ -540,6 +551,46 @@ public void testSlotReleaseAtomicallyReleasesExecution() throws Exception {
 		});
 	}
 
+	/**
+	 * Tests that producedPartitions are registered synchronously under an asynchronous interface.
+	 */
+	@Test(expected = IllegalStateException.class)
```

Comment:
I prefer `try/catch` blocks since they guarantee the exception occurs where we want it to.

Reply:
by `try/catch`

I guess you mean something like this?
```
    boolean incompletePartitionRegistrationFuture = false;
		try {
			Execution.registerProducedPartitions(sourceVertex, new LocalTaskManagerLocation(), new ExecutionAttemptID(), false);
		} catch (IllegalStateException e) {
			incompletePartitionRegistrationFuture = true;
		}

		assertTrue(incompletePartitionRegistrationFuture);
```

Reply:
```
try {
	Execution.registerProducedPartitions(sourceVertex, new LocalTaskManagerLocation(), new ExecutionAttemptID(), false);
	Assert.fail("expected exception");
} catch (IllegalStateException e) {
	// expected
}
```

Reply:
I see, thanks!